code/modules/mob/living/carbon/hallucination.dm

```diff
@@ -0,0 +1,1102 @@
+var/list/hallucinated_phrases = file2list("config/hallucination/hallucinated_phrases.txt")
+var/list/hallucinated_actions = file2list("config/hallucination/hallucinated_actions.txt")	//important note when adding to this file: "you" will always be replaced by the hallucinator's name
+var/list/hallucinated_thoughts = file2list("config/hallucination/hallucinated_thoughts.txt")
+
+
+/mob/living/carbon/var/next_hallucination = 0			//Hallucination spam limit var
+/mob/living/carbon/var/list/hallucinations = list()		//Hallucinations currently affecting the mob. Not to be confused with singular "hallucination" which is a NUM variable like confused/drowsy/eye_blind etc
+
+///////////////////////////////////////Hallucinated Hearing///////////////////////////
+/mob/living/carbon/hear_say(var/message, var/verb = "says", var/datum/language/language = null, var/alt_name = "",var/italics = 0, var/mob/speaker = null, var/sound/speech_sound, var/sound_vol)
+	if(hallucination >= 60 && prob(1))
+		var/orig_message = message
+		message = LAZYPICK(hallucinated_phrases, message)
+		log_say("Hallucination level changed [orig_message] by [speaker] to [message] for [key_name(src)].", ckey=key_name(src))
+	..()
+/mob/living/carbon/hear_radio(var/message, var/verb="says", var/datum/language/language=null, var/part_a, var/part_b, var/mob/speaker = null, var/hard_to_hear = 0, var/vname ="")
+	if(hallucination >= 60 && prob(1))
+		var/orig_message = message
+		message = LAZYPICK(hallucinated_phrases, message)
+		log_say("Hallucination level changed [orig_message] by [speaker] to [message] for [key_name(src)].", ckey=key_name(src))
+	..()
+
+/mob/living/carbon/proc/handle_hallucinations()     //Main handling proc, called in life()
+
+	hallucination -= 1	//Tick down the duration
+
+	if(!hallucination)  //We're done
+		return
+
+	if(!client || stat || world.time < next_hallucination)
```

Comment:
If they're unconscious their hallucinations don't tick?

Reply:
Hallucination ticks down before the stat check, on line 25. It ticks down but they do not experience hallucination effects